### PR TITLE
drivers: i2c: i2c_nrfx_twi: Add return -EBUSY when I2C bus is busy

### PR DESCRIPTION
**Pull request by AurelienV-ProGlove**
Return -EBUSY instead of -EIO when trying to access 
the I2C bus while it is currently used

Signed-off-by: Aurelien Vouaillat <aurelien.vouaillat@proglove.de>

### CODE REVIEW

File: drivers/i2c/i2c_nrfx_twi.c [hunk 1]
```diff
@@ -53,7 +53,11 @@ static int i2c_nrfx_twi_transfer(struct device *dev, struct i2c_msg *msgs,
 					       (msgs[i].flags & I2C_MSG_STOP) ?
 					       0 : NRFX_TWI_FLAG_TX_NO_STOP);
 		if (res != NRFX_SUCCESS) {
-			return -EIO;
+			if (res == NRFX_ERROR_BUSY) {
+				return -EBUSY;
```
Comment (tbursztyka) [inline drivers/i2c/i2c_nrfx_twi.c]: Not related to that PR, but that reminds me i²c API isn't very nice with the user as it does not provide the same thing as spi_context_lock() for concurrent access. (But both of the API do not provide timeout, which another issue).

**Comment by nashif** [comment]
please fix the commit message:

1: UC5 Title exceeds max length (75>72): "drivers: i2c: i2c_nrfx_twi: Add return -EBUSY when I2C bus is in busy state"
3: UC4 Line exceeds max length (90>72): "Return -EBUSY instead of -EIO when trying to access the I2C bus while it is currently used"

**Review by Mierunski — CHANGES_REQUESTED** [review]
Please make the same change for i2c_nrfx_twim.c

**Comment by AurelienV-ProGlove** [comment]
Added the modification to i2c_nrfx_twim as well and reduced the length of commits messages

**Review by carlescufi — APPROVED** [review]

**Review by Mierunski — APPROVED** [review]

**Pull request merged**